### PR TITLE
feat(providers): add ability to register tax and fulfillment providers using medusa-extender annotations

### PR DESCRIPTION
**Pull request by dwene**

### CODE REVIEW

**Comment by dwene** [comment]
@adrien2p ok I added some unit and integration tests! I'd like to start with these 2 fulfillment providers for now, but I'm setting up patterns here to be used again for payment, search, notification, file, oauth,  and other future providers!

**Comment by dwene** [comment]
Also this closes #128

File: src/decorators/providers.decorator.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { ProviderInjectableOptions } from '../core';
+import { buildProviderDecorator } from './helpers/build-provider-decorator';
+
+export type AllowedProviderInjectableOptions = Omit<ProviderInjectableOptions, 'subtype'>;
+/**
+ * Mark a class as an entity to be used by the loader.
```
Comment (adrien2p) [inline src/decorators/providers.decorator.ts:6]: todo: update

File: src/loaders/plugins.loader.ts [hunk 2]
```diff
@@ -1,20 +1,27 @@
 import { Express, NextFunction, Response } from 'express';
 import { MedusaAuthenticatedRequest, MedusaContainer, MedusaRequest } from './';
-import { customEventEmitter } from '../core';
+import { customEventEmitter, GetInjectableOptions } from '../core';
+import { registerProviders } from './providers.loader';
+import { container } from 'webpack';
```
Comment (adrien2p) [inline src/loaders/plugins.loader.ts:5]: todo: rm

**Review by adrien2p — CHANGES_REQUESTED** [review]
Looks very nice, Can I get you to resolve the codeQL comment as well and mine and it should be ready to merge 🚀

**Review by adrien2p — APPROVED** [review]
LGTM 🔥

**Comment by dwene** [comment]
Sweet! Let's ship it!

**Comment by adrien2p** [comment]
@dwene i ve merged it. Can I get you to update the doc please?

**Pull request merged**